pkg/k8shandler/service_monitor.go

```diff
@@ -0,0 +1,76 @@
+package k8shandler
+
+import (
+	"fmt"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/client/monitoring/v1"
+	v1alpha1 "github.com/openshift/elasticsearch-operator/pkg/apis/elasticsearch/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CreateOrUpdateServiceMonitors ensures the existence of ServiceMonitors for Elasticsearch cluster
+func CreateOrUpdateServiceMonitors(dpl *v1alpha1.Elasticsearch) error {
+	serviceMonitorName := fmt.Sprintf("monitor-%s-%s", dpl.Name, "cluster")
+	owner := asOwner(dpl)
+
+	labelsWithDefault := appendDefaultLabel(dpl.Name, dpl.Labels)
+
+	err := createOrUpdateServiceMonitor(serviceMonitorName, dpl.Namespace, labelsWithDefault, owner)
+	if err != nil {
+		return fmt.Errorf("Failure creating service %v", err)
+	}
+
+	return nil
+}
+
+func createOrUpdateServiceMonitor(serviceMonitorName, namespace string, labels map[string]string, owner metav1.OwnerReference) error {
+	elasticsearchScMonitor := createServiceMonitor(serviceMonitorName, namespace, labels)
+	addOwnerRefToObject(elasticsearchScMonitor, owner)
+	err := sdk.Create(elasticsearchScMonitor)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("Failure constructing Elasticsearch ServiceMonitor: %v", err)
+	}
+
+	// TODO: handle update
+
+	return nil
+}
+
+func createServiceMonitor(serviceMonitorName, namespace string, labels map[string]string) *monitoringv1.ServiceMonitor {
+	svcMonitor := serviceMonitor(serviceMonitorName, namespace, labels)
+	labelSelector := metav1.LabelSelector{
+		MatchLabels: labels,
+	}
+	tlsConfig := monitoringv1.TLSConfig{
+		CAFile: "/etc/prometheus/configmaps/prometheus-serving-certs-ca-bundle/service-ca.crt",
+	}
+	endpoint := monitoringv1.Endpoint{
+		Port:            "restapi",
+		Path:            "/_prometheus/metrics",
+		Scheme:          "https",
+		BearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token",
+		TLSConfig:       &tlsConfig,
+	}
+	svcMonitor.Spec = monitoringv1.ServiceMonitorSpec{
+		JobLabel:  "monitor-elasticsearch",
+		Endpoints: []monitoringv1.Endpoint{endpoint},
+		Selector:  labelSelector,
+	}
+	return svcMonitor
+}
+
+func serviceMonitor(serviceMonitorName string, namespace string, labels map[string]string) *monitoringv1.ServiceMonitor {
+	return &monitoringv1.ServiceMonitor{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       monitoringv1.ServiceMonitorsKind,
+			APIVersion: monitoringv1.Group + "/" + monitoringv1.Version,
```

Comment:
`monitoringv1.SchemeGroupVersion.String()`

Reply:
Doesnt exist. But for the record, I made a bet that I'll buy @ewolinetz 10 beers if it does. And I encourage PRs ;-)
@jcantrill is our witness here!